### DISCUSSION TITLE
咨询一下Android端的PAGText和PAGTextLayer用法

### DISCUSSION THREAD

**Discussion by zgglin** [Tencent/libpag#1699] [Q&A] [answered]
咨询一下Android端的PAGText和PAGTextLayer：
使用版本4.2.100
我从pagFile.getTextData(editableIndex)中获取到PAGText，设置fillColor后通过pagFile.replaceText(editableIndex, pagText)使颜色修改生效。
1. 再次调用pagFile.getTextData(editableIndex)获取PAGText，此时里面的fillColor不是我上一步所设置的颜色值。其他属性值也都是默认值 -- 请问是设计如此吗？
2. 通过pagTextLayer.fillColor() 可以读取到 pagFile.replaceText所设置的颜色值。--  PAGText中的其他属性如backgroundColor无法通过PagTextLayer获取到。

**Comment by kevingpqi123** [comment, marked as answer]
1、返回的数据 API 文档中已经有注释，可以看下 API 文档：
public [PAGText](https://pag.art/apis/android/reference/org/libpag/PAGText.html) getTextData (int index)
Get a text data of the specified index. The index ranges from 0 to numTexts - 1. Note: It always returns the default text data.

2、文本的背景颜色不支持外部设置，因此 PAGTextLayer 中没有相关接口
